lib/piculet/exporter.rb

```diff
@@ -37,27 +36,32 @@ def export
     private
     def export_security_group(security_group)
       {
-        :name        => security_group.name,
+        :name        => security_group.group_name,
         :description => security_group.description,
         :tags        => tags_to_hash(security_group.tags),
         :owner_id    => security_group.owner_id,
-        :ingress     => export_ip_permissions(security_group.ingress_ip_permissions),
-        :egress      => export_ip_permissions(security_group.egress_ip_permissions),
+        :ingress     => export_ip_permissions(security_group.ip_permissions),
+        :egress      => export_ip_permissions(security_group.ip_permissions_egress),
       }
     end
 
     def export_ip_permissions(ip_permissions)
-      ip_permissions = ip_permissions ? ip_permissions.aggregate : []
+      ip_permissions = ip_permissions || []
 
       ip_permissions = ip_permissions.map do |ip_perm|
+        ip_protocol = ip_perm.ip_protocol == "-1" ? :any : ip_perm.ip_protocol.to_sym
+        port_range = ip_perm.from_port..ip_perm.to_port
+        port_range = nil if port_range == (nil..nil)
+        ip_ranges = ip_perm.ip_ranges.map { |range| range.cidr_ip }.sort
         {
-          :protocol   => ip_perm.protocol,
-          :port_range => ip_perm.port_range,
-          :ip_ranges  => ip_perm.ip_ranges.sort,
-          :groups => ip_perm.groups.map {|group|
+          :protocol   => ip_protocol,
+          :port_range => port_range,
+          :ip_ranges  => ip_ranges,
+          :groups => ip_perm.user_id_group_pairs.map {|group|
+            group = @ec2.security_groups.find { |g| g.id == group.group_id }
```

Comment:
L61's `group` and L63's `group` is different.

And L61's `group` may be `nil`.

I try `bundle exec bin/piculet -e` on my environment . So, raise following error.

```
[Aws::EC2::Client 200 0.186472 0 retries] describe_security_groups()

bundler: failed to load command: bin/piculet (bin/piculet)
NoMethodError: undefined method `group_id' for nil:NilClass
  /Users/k1low/.ghq/github.com/codenize-tools/piculet/lib/piculet/exporter.rb:63:in `block (2 levels) in export_ip_permissions'
  /Users/k1low/.ghq/github.com/codenize-tools/piculet/lib/piculet/exporter.rb:60:in `map'
  /Users/k1low/.ghq/github.com/codenize-tools/piculet/lib/piculet/exporter.rb:60:in `block in export_ip_permissions'
  /Users/k1low/.ghq/github.com/codenize-tools/piculet/lib/piculet/exporter.rb:51:in `map'
  /Users/k1low/.ghq/github.com/codenize-tools/piculet/lib/piculet/exporter.rb:51:in `export_ip_permissions'
  /Users/k1low/.ghq/github.com/codenize-tools/piculet/lib/piculet/exporter.rb:43:in `export_security_group'
  /Users/k1low/.ghq/github.com/codenize-tools/piculet/lib/piculet/exporter.rb:30:in `block in export'
  /Users/k1low/.ghq/github.com/codenize-tools/piculet/lib/piculet/exporter.rb:22:in `each'
  /Users/k1low/.ghq/github.com/codenize-tools/piculet/lib/piculet/exporter.rb:22:in `export'
  /Users/k1low/.ghq/github.com/codenize-tools/piculet/lib/piculet/exporter.rb:5:in `export'
  /Users/k1low/.ghq/github.com/codenize-tools/piculet/lib/piculet/client.rb:41:in `export'
  bin/piculet:182:in `<top (required)>'
```

Reply:
Thanks for your review!

I cannot immediately come up with why `@ec2.security_groups.find { |g| g.id == group.group_id }` returns `nil`, so I try to search the cause.

I added the L61 `@ec2.security_groups.find { |g| g.id == group.group_id }` to fetch `group.group_name` info. Because `ip_perm.user_id_group_pair`'s group (L60) has always no information about `group_name` as below.

```
    57:           :protocol   => ip_protocol,
    58:           :port_range => port_range,
    59:           :ip_ranges  => ip_ranges,
    60:           :groups => ip_perm.user_id_group_pairs.map {|group|
    61:             require 'pry'
 => 62:             binding.pry
    63:             group = @ec2.security_groups.find { |g| g.id == group.group_id }
    64:             {
    65:               :id       => group.group_id,
    66:               :name     => group.group_name,
    67:               :owner_id => group.owner_id,

[1] 2.4.1-p111(#<Piculet::Exporter>)> group
=> #<struct Aws::EC2::Types::UserIdGroupPair
 description=nil,
 group_id="sg-xxx",
 group_name=nil, # here!
 peering_status=nil,
 user_id="752xxxxxxxxx",
 vpc_id=nil,
 vpc_peering_connection_id=nil>
[2] 2.4.1-p111(#<Piculet::Exporter>)>
```